builder/simplify.cc

```diff
@@ -323,11 +333,11 @@ class simplifier : public node_mutator {
     // - Prune dead lets
     // - Inline single-ref lets, along with lets that are just constants or vars
     for (auto it = lets.begin(); it != lets.end();) {
-      int refs = *references[it->first];
-      if (refs == 0) {
+      auto& ref_info = references[it->first];
+      if (!ref_info || ref_info->counter_ == 0) {
```

Comment:
This should not need to check if `ref_info` is set, we should just assert that. If it's missing, something has gone very wrong (it's set above).